File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        maxArea = -inf

        left = 0
        right = len(height)-1


        while left < right:
            minHeight = min(height[left], height[right])
            maxArea = max(maxArea, minHeight*(right-left))            

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return maxArea

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {

        int left = 0;
        int right = height.length - 1;
        int maxAmount = 0;

        while(left < right){
            int currHeight = Math.min(height[left], height[right]);
            maxAmount = Math.max(maxAmount, (right-left) * currHeight);

            if(height[left] > height[right]){
                right--;
            } else {
                left++;
            }
        }

        return maxAmount;        
    }
}